### DISCUSSION TITLE
When SO becoming ECS compliant?

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [Security-Onion-Solutions/securityonion#4430] [Unsupported Versions] [answered]
I've seen in docs Security Onion is in transition to become Elastic Common Schema (ECS) compliant, When we can expect that?

**Comment by TOoSmOotH** [comment, marked as answer]
https://github.com/Security-Onion-Solutions/securityonion/discussions/4434

**Reply by Jackson-Pollock** [reply]
ECS not yet in this release? And incase SO become ECS compliant, then all old queries (which has old fields name) will break right?

**Reply by dougburks** [reply]
@TOoSmOotH answered this question over on the other thread:
https://github.com/Security-Onion-Solutions/securityonion/discussions/4434#discussioncomment-840277

**Comment by rwaight** [comment]
@TOoSmOotH is there a plan to define ECS compliant mappings in the [so-common-template](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/templates/so/so-common-template.json) in order to prevent mapping conflicts?  If not, is there a way to volunteer to help work on making SO more compliant with ECS?

I recently deployed SO 2.3.61 and immediately found mapping issues with `process.pid`, `event.duration`, and `event.module`. 
 I was hoping to update the mappings for the `so-common` Elasticseach index template, but found that if I defined `event.module` (within the [`event` object](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/templates/so/so-common-template.json#L229-L232)) as a `keyword` (as defined in the ECS documentation) that I am unable to look at any `event.module` fields in the Security Onion Dashboard as the Kibana saved object is looking for the `event.module.keyword` field instead.

On that note, I noticed that the [kibana saved objects file](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/kibana/files/saved_objects.ndjson.jinja) has references to many ECS fields with `.keyword` at the end of the fields.  It looks like updates would be need to be made to both the `so-common-template.json` file as well as the `saved_objects.ndjson.jinja` file.

**Reply by weslambert** [reply]
I'm curious where there were issues with `event.module`.

Are you referring to having event.module itself be typed as keyword, as opposed to using a subfield/separately indexed field?

**Reply by rwaight** [reply]
I found the issue with `event.module` when I specified the field as a keyword. I found that the Kibana SO dashboard was looking for the field `event.module.keyword`, so no data was populated in the SO module visualization.

> Are you referring to having event.module itself be typed as keyword, as opposed to using a subfield/separately indexed field?

Yes, if `event.module` is defined as `type: keyword` then Elasticsearch would not create a `event.module.keyword` field, which would align with the [`event.module` field ECS definition](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-module).

I'd be willing to open a GH issue as well as pull request to propose changes to the `event` fieldset fields and the Kibana dashboards, just let me know if that is the appropriate route for this.
